drivers/wireless/bluetooth/bt_rpmsghci_server.c

```diff
@@ -0,0 +1,547 @@
+ * Name: rpmsghci_ns_bind
+ *
+ * Description:
+ *   Bind to the rpmsg name service.
+ *
+ ****************************************************************************/
+
+static void rpmsghci_ns_bind(FAR struct rpmsg_device *rdev, FAR void *priv,
+                             FAR const char *name, uint32_t dest)
+{
+  FAR struct rpmsghci_server_s *server = priv;
+
+  server->ept.priv = priv;
+  rpmsg_create_ept(&server->ept, rdev, name,
+                   RPMSG_ADDR_ANY, dest,
+                   rpmsghci_ept_cb, rpmsghci_ns_unbind);
+}
+
+/****************************************************************************
+ * Name: rpmsghci_bt_receive
+ *
+ * Description:
+ *   Called by the Bluetooth low-level driver when new data is received from
+ *   the radio.  Transfers data to the RPMSG client.
+ *
+ *   This callback should not be called from the interrupt context !
+ *
+ * Returned Value:
+ *   OK on success; A negated errno value is returned on any failure.
+ *
+ ****************************************************************************/
+
+int rpmsghci_bt_receive(FAR struct bt_driver_s *btdev,
+                        enum bt_buf_type_e type, FAR void *data, size_t len)
+{
+  FAR struct rpmsghci_server_s *priv  = NULL;
+  FAR struct rpmsghci_data_s   *msg   = NULL;
+  uint32_t                      space = 0;
+  int                           ret   = OK;
+
+  wlinfo("rpmsghci_bt_receive %d\n", type);
+
+  /* Get RPMSG-HCI data */
+
+  DEBUGASSERT(btdev != NULL);
+  priv = btdev->priv;
+
+  if (data == NULL)
+    {
+      ret = -EINVAL;
+      goto errout;
+    }
+
+  /* Perform the rpmsg write */
+
+  msg = rpmsghci_get_tx_payload_buffer(priv, &space);
+  if (msg == NULL)
+    {
+      ret = -ENOMEM;
+      goto errout;
+    }
+
+  DEBUGASSERT(sizeof(struct rpmsghci_data_s) - 1 + len <= space);
+  memcpy(msg->data, data, len);
+  msg->type = type;
+  ret = rpmsghci_send(priv, RPMSGHCI_RECV, &msg->header,
+                      sizeof(struct rpmsghci_data_s) -1 + len);
+
+errout:
+  return ret;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: rpmsghci_server_init
+ *
+ * Description:
+ *   Rpmsg-HCI server initialize function, the server cpu should call
+ *   this function.
+ *
+ * Parameters:
+ *   name - RPMSG-HCI server name
+ *   bt   - BT device handler
+ *
+ * Returned Values:
+ *   OK on success; A negated errno value is returned on any failure.
+ *
+ ****************************************************************************/
+
+int rpmshci_server_init(FAR const char *name, FAR struct bt_driver_s *btdev)
+{
+  struct rpmsghci_server_s *priv = NULL;
+  int                       ret  = OK;
+
+  /* Validate input */
+
+  if (!name || !btdev)
+    {
+      ret = -EINVAL;
```

Comment:
```suggestion
      return -EINVAL;
```
and remove errout label